diag_manager/fms_diag_object.F90

```diff
@@ -396,17 +422,323 @@ function diag_obj_is_static (obj) result (rslt)
     rslt = obj%static
 end function diag_obj_is_static
 
-function diag_obj_get_vartype (obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    integer :: rslt
-    rslt = obj%vartype
-end function diag_obj_get_vartype
-
-function diag_obj_get_varname(obj) result (rslt)
-    class(fmsDiagObject_type), intent(in) :: obj
-    character(len=len(obj%varname)) :: rslt
-    rslt = obj%varname
-end function diag_obj_get_varname
+!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+!! Get functions
+
+!> @brief Gets metedata
+!! @return metadata string array, or a single space if metadata is not allocated
+function get_metadata (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     character(len=:), allocatable, dimension(:) :: rslt
+     if (allocated(obj%metadata)) then
+       allocate(character(len=(len(obj%metadata(1)))) :: rslt (size(obj%metadata)) )
+       rslt = obj%metadata
+     else
+       allocate(character(len=1) :: rslt(1:1))
+       rslt = " "
+     endif
+end function get_metadata
+!> @brief Gets static
+!! @return true if the variable is static
+function get_static (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     logical :: rslt 
+     rslt = obj%static
+end function get_static
+!> @brief Gets regisetered
+!! @return true if the object is registered
+function get_registered (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     logical :: rslt 
+     rslt = obj%registered
+end function get_registered
+!> @brief Gets mask variant
+!! @return true if there is a mask variant
+function get_mask_variant (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     logical :: rslt 
+     rslt = obj%mask_variant
+end function get_mask_variant
+!> @brief Gets local
+!! @return true if the variable is a local variable 
+function get_local (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     logical :: rslt 
+     rslt = obj%local
+end function get_local
+!> @brief Gets initial time 
+!! @return the initial time
+!! TODO
+!function get_init_time (obj) &
+!result(rslt)
+!     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+!     TYPE(time_type) :: rslt 
+!
+!end function get_init_time
+!> @brief Gets vartype 
+!! @return  The integer related to the variable type
+function get_vartype (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
+     integer :: rslt 
+     rslt = obj%vartype
+end function get_vartype
+!> @brief Gets varname
+!! @return the variable name
+function get_varname (obj) &
+result(rslt)
+     class (fmsDiagObject_type), intent(in) :: obj !< diag object
```

Comment:
I guess in this case the getter litterally returns the member field (seems to pass the litmus test). But I will read (or as) what exatly does the assignemnt (```=```) do.  Hope we dont have to look at assemberr  :-)